File: pinyougou-web-user/src/main/webapp/js/controller/addressController.js
//地址控制器
// 定义控制器:
app.controller("addressController",function($scope,$controller,$http,addressService) {

    $scope.entity = {};
    // 查询所有的品牌列表的方法:
    $scope.findListByLoginUser = function () {
        // 向后台发送请求:
        addressService.findListByLoginUser().success(function (response) {
            $scope.addressList = response;
        });
    }
    //新增地址
    $scope.addAddress = function(){
        // 区分是保存还是修改
        var object;
        if($scope.entity.id != null){
            // 更新

            object = addressService.update($scope.entity);
        }else{
            // 保存
            $scope.entity = {};
            object = addressService.add($scope.entity);
        }
        object.success(function(response){
            // {flag:true,message:xxx}
            // 判断保存是否成功:
            if(response.flag){
                //重新查询
                alert(response.message);
                $scope.entity = {};
                location.href="home-setting-address.html";
            }else{
                // 保存失败
                alert(response.message);
            }
        });
    }


    /*$scope.addAddress = function () {
        // 向后台发送请求:
        addressService.addAddress($scope.entity).success(function(response){
            if(response.flag){
                //重新查询
                alert(response.message);
                location.href="home-setting-address.html";
            }else{
                alert(response.message);
            }

        });
    }*/

    // 删除地址:
    $scope.dele = function(id){
        addressService.dele(id).success(function(response){
            // 判断保存是否成功:
            if(response.flag){
                // 保存成功
                // alert(response.message);
                alert(response.message);
                location.href="home-setting-address.html";
            }else{
                // 保存失败
                alert(response.message);
            }
        });
    }
    // 删除地址:
    $scope.setDefault = function(id){
        addressService.setDefault(id).success(function(response){
            // 判断保存是否成功:
            if(response.flag){
                // 保存成功
                // alert(response.message);
                alert(response.message);
                location.href="home-setting-address.html";
            }else{
                // 保存失败
                alert(response.message);
            }
        });
    }

    // 查询一个地址:
    $scope.findOneById = function(id){
        addressService.findOneById(id).success(function(response){
            $scope.entity = response;
        });
    }


});

File: pinyougou-web-user/src/main/java/cn/itcast/core/controller/AddressController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.user.User;
import cn.itcast.core.service.AddressService;
import com.alibaba.dubbo.config.annotation.Reference;

import entity.Result;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;
import java.util.List;

@RestController
@RequestMapping("address")
/**
 * 地址管理
 */
public class AddressController {

    @Reference
    private AddressService addressService;

    //"../address/findAll.do"
    @RequestMapping("findListByLoginUser")
    public List<Address> findListByLoginUser() {
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        return addressService.findListByLoginUser(name);
    }

    //新增地址
    @RequestMapping("addAddress")
    public Result addAddress(@RequestBody Address address) {
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        address.setUserId(name);
        address.setCreateDate(new Date());
        //设置是否默认
        address.setIsDefault("0");
        try {
            addressService.addAddress(address);
            return new Result(true, "保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "保存失败");
        }

    }
    @RequestMapping("update")
    public Result update(@RequestBody Address address) {

        address.setCreateDate(new Date());
        try {
            addressService.update(address);
            return new Result(true, "保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "保存失败");
        }

    }
    //get("../address/delete.do?id="+id);
    //删除地址
    @RequestMapping("delete")
    public Result delete(Long id) {

        try {
            addressService.delete(id);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }

    }
    //设为默认地址
    //get("../address/setDefault.do?id="+id);
    @RequestMapping("setDefault")
    public Result setDefault(Long id) {
        String name = SecurityContextHolder.getContext().getAuthentication().getName();

        try {
            addressService.setDefault(id,name);
            return new Result(true, "设置成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "设置失败");
        }

    }

    //get("../address/findOneById.do?id="+id);
    //根据id查询一个地址
    @RequestMapping("findOneById")
    public Address findOneById(Long id) {
        return addressService.findOneById(id);
    }

}
